packages/components/inputs/search-select-input/src/search-select-input.js

```diff
@@ -0,0 +1,151 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import { ThemeContext } from '@emotion/core';
+import { SafeHTMLElement } from '@commercetools-uikit/utils';
+import { SearchIconDropdownIndicator } from '@commercetools-uikit/select-utils';
+import AsyncSelectInput from '@commercetools-uikit/async-select-input';
+import messages from '../../../../../src/components/internals/messages/select';
+import createSelectStyles from '../../../../../src/components/internals/create-select-styles';
+
+const useTheme = () => useContext(ThemeContext);
+
+const SearchSelectInput = (props) => {
+  const intl = useIntl();
+  const theme = useTheme();
+  const noOptionsMessageWithOutInputValue = props.initialSuggestionMessage
+    ? props.initialSuggestionMessage()
+    : intl.formatMessage(messages.noOptionsMessageWithoutInputValue);
+  const noOptionsMessageWithInputValue = props.noOptionsMessage
+    ? props.noOptionsMessage()
+    : intl.formatMessage(messages.noOptionsMessageWithInputValue);
+
+  return (
+    <AsyncSelectInput
+      horizontalConstraint={props.horizontalConstraint}
+      // react-select uses "id" (for the container) and "inputId" (for the input),
+      // but we use "id" (for the input) and "containerId" (for the container)
+      // instead.
+      // This makes it easier to less confusing to use with <label />s.
+      id={props.containerId}
+      inputId={props.id}
+      aria-label={props['aria-label']}
+      aria-labelledby={props['aria-labelledby']}
+      name={props.name}
+      placeholder={props.placeholder}
+      tabIndex={props.tabIndex}
+      iconLeft={props.iconLeft}
+      inputValue={props.inputValue}
+      value={props.value}
+      backspaceRemovesValue={props.backspaceRemovesValue}
+      hasWarning={props.hasWarning}
+      hasError={props.hasError}
+      isDisabled={props.isDisabled || props.isReadOnly}
+      isClearable={props.isClearable}
+      isOptionDisabled={props.isOptionDisabled}
+      isMulti={props.isMulti}
+      isSearchable={true}
+      autoFocus={props.isAutofocussed}
+      components={{
+        ...props.components,
+        DropdownIndicator: SearchIconDropdownIndicator,
+      }}
+      noOptionsMessage={({ inputValue }) =>
+        inputValue === ''
+          ? noOptionsMessageWithOutInputValue
+          : noOptionsMessageWithInputValue
+      }
+      styles={createSelectStyles(
+        {
+          menuPortalZIndex: props.menuPortalZIndex,
```

Comment:
If this needed if we pass the `menuPortalZIndex` prop and let the underlying component do it itself?